--- libxkbcommon-1.6.0/include/xkbcommon/xkbcommon-keysyms.h~	2023-10-08 20:46:48.000000000 +0000
+++ libxkbcommon-1.6.0/include/xkbcommon/xkbcommon-keysyms.h	2023-10-12 18:36:12.911087853 +0000
@@ -439,6 +439,12 @@
 #define XKB_KEY_dead_belowcomma               0xfe6e
 #define XKB_KEY_dead_currency                 0xfe6f
 
+/* extra dead elements for German T3 layout */
+#define XKB_KEY_dead_lowline                  0xfe90
+#define XKB_KEY_dead_aboveverticalline        0xfe91
+#define XKB_KEY_dead_belowverticalline        0xfe92
+#define XKB_KEY_dead_longsolidusoverlay       0xfe93
+
 /* dead vowels for universal syllable entry */
 #define XKB_KEY_dead_a                        0xfe80
 #define XKB_KEY_dead_A                        0xfe81
